sidecar/supervisor/appsupervisor.go

```diff
@@ -64,11 +77,7 @@ func (a *AppSupervisor) DoAppSupervision(cmdArgs []string, agent *register.Regis
 		select {
 		case sig := <-sigChan:
 			log.Infof("Intercepted signal '%s'", sig)
-			if appProcess != nil {
-				appProcess.Signal(sig)
-			} else {
-				a.Shutdown(0)
-			}
+			a.Shutdown(0)
```

Comment:
Need to check if its the app process or a helper process. Only the app process dies, then we kill all other processes.

The current implementation will kill the container even if a helper process like filebeat dies.